geoutils/georaster/raster.py

```diff
@@ -559,12 +598,17 @@ def _overloading_check(
         # Case 2 - other is a numpy array
         elif isinstance(other, np.ndarray):
             # Check that both array have the same shape
-            if self.data.shape == other.shape:
+
+            if len(other.shape) == 2:
```

Comment:
There's a function for getting the dimension of the array: `np.ndim`. 
Of course here is does not matter, but it works also for lists/tuples.

Reply:
OK. Here for consistency I copied the syntax that was already used in `data.setter`